rpc/amqp-rpc.go

```diff
@@ -581,11 +590,15 @@ func (rpc *AmqpRPCCLient) DispatchSync(method string, body []byte) (response []b
 		}
 		err = unwrapError(rpcResponse.Error)
 		if err != nil {
+			rpc.stats.Inc(fmt.Sprintf("RPC.Latency.%s.Error", method), 1, 1.0)
 			return
 		}
+		rpc.stats.Inc("RPC.Rate.Success", 1, 1.0)
+		rpc.stats.TimingDuration(fmt.Sprintf("RPC.Latency.%s.Success", method), time.Since(callStarted), 1.0)
 		response = rpcResponse.ReturnVal
 		return
 	case <-time.After(rpc.timeout):
+		rpc.stats.Inc("RPC.Rate.Timeouts", 1, 1.0)
```

Comment:
This is all much easier to grok.

Latency calculations need to include timeouts or else they look way better than they are.

Having a Latency.%s.Success and a Latency.%s should suffice.